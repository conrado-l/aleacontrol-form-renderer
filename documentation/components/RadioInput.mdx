---
name: RadioInput
---

import { Playground, PropsTable } from 'docz'
import RadioInput from '../../src/components/RadioInput.js'

# RadioInput

<PropsTable of={RadioInput} />

## Description
A radio input with options
## Demo and usage

<Playground>
    <RadioInput value='male' name='gender' options={[{value: 'male', description: 'Male'}, {value: 'female', description: 'Female'}]} col={4} update={(e) => e}/>
</Playground>
