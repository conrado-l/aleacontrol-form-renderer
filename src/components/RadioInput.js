import React, { Component } from 'react'
import PropTypes from 'prop-types'

class RadioInput extends Component {
  render () {
    // Generate the radio button options
    const buttons = this.props.options.map((input) => {
      return <div key={input.value}>
        <input type='radio' id={input.value} name={this.props.name} value={input.value} />
        <label htmlFor={input.value}> {input.description} </label>
      </div>
    })
    return (
      <div onChange={(e) => this.props.update(this.props.name, e.target.value)}>
        {buttons}
      </div>
    )
  }
}

RadioInput.propTypes = {
  options: PropTypes.array,
  name: PropTypes.string,
  update: PropTypes.func
}
export default RadioInput
