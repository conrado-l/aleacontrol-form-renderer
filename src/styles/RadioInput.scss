@import "mixins";

.radio-input-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
}

@include mobile() {
  .radio-input-container {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
}
