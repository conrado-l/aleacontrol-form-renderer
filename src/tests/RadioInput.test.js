import React from 'react'
import { mount } from 'enzyme'
import RadioInput from '../components/RadioInput'
import renderer from 'react-test-renderer'

describe('RadioInput', () => {
  const radioOptions = [
    { value: 'male', description: 'Male' },
    { value: 'female', description: 'Female' }
  ]

  it('should render correctly radio component', () => {
    const RadioInputComponent = renderer
      .create(<RadioInput name='gender' value={radioOptions[0].value} options={radioOptions} />)
      .toJSON()
    expect(RadioInputComponent).toMatchSnapshot()
  })

  it('should render the radio options', () => {
    const RadioInputComponent = mount(<RadioInput name='gender' value={radioOptions[0].value}
      options={radioOptions} />)

    expect(RadioInputComponent.containsMatchingElement(
      <input type='radio' name={RadioInputComponent.props().name} id={radioOptions[0].value}
        value={radioOptions[0].value} defaultChecked />)).toBeTruthy()

    expect(RadioInputComponent.containsMatchingElement(
      <label htmlFor={radioOptions[0].value}> {radioOptions[0].description} </label>)).toBeTruthy()

    expect(RadioInputComponent.containsMatchingElement(
      <input type='radio' name={RadioInputComponent.props().name} id={radioOptions[1].value}
        value={radioOptions[1].value} />)).toBeTruthy()

    expect(RadioInputComponent.containsMatchingElement(
      <label htmlFor={radioOptions[1].value}> {radioOptions[1].description} </label>)).toBeTruthy()
  })

  it('should call the update callback function with the proper values on change', () => {
    const changFn = jest.fn()
    const RadioInputComponent = mount(<RadioInput name='gender' value={radioOptions[0].value} options={radioOptions}
      update={changFn} />)

    RadioInputComponent.find(`input#${RadioInputComponent.props().options[1].value}`).simulate('change', {
      target: {
        value: RadioInputComponent.props().options[1].value
      }
    })

    expect(changFn).toHaveBeenCalled()

    expect(changFn).toHaveBeenCalledWith(RadioInputComponent.props().name, RadioInputComponent.props().options[1].value)
  })
})
