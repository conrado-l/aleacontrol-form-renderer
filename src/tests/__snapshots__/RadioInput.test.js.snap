// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioInput should render correctly radio component 1`] = `
<div
  onChange={[Function]}
>
  <div>
    <input
      defaultChecked={true}
      id="male"
      name="gender"
      type="radio"
      value="male"
    />
    <label
      htmlFor="male"
    >
       
      Male
       
    </label>
  </div>
  <div>
    <input
      defaultChecked={false}
      id="female"
      name="gender"
      type="radio"
      value="female"
    />
    <label
      htmlFor="female"
    >
       
      Female
       
    </label>
  </div>
</div>
`;
